client/src/components/client-dashboard-component/FilterCard.jsx

```diff
@@ -0,0 +1,226 @@
+import { useState } from "react";
+import {
+  Box,
+  TextField,
+  Checkbox,
+  FormControlLabel,
+  Radio,
+  RadioGroup,
+  Button,
+  MenuItem,
+  Select,
+  Card,
+  CardContent,
+  Typography,
+  InputAdornment,
+} from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search";
+
+function FilterCard() {
+
+  const dateNow = new Date(); // Creating a new date object with the current date and time
+  const year = dateNow.getFullYear(); // Getting current year from the created Date object
+  const monthWithOffset = dateNow.getUTCMonth() + 1; // January is 0 by default in JS. Offsetting +1 to fix date for calendar.
+  const month = // Setting current Month number from current Date object
+    monthWithOffset.toString().length < 2 // Checking if month is < 10 and pre-prending 0 to adjust for date input.
+      ? `0${monthWithOffset}`
+      : monthWithOffset;
+  const date =
+    dateNow.getUTCDate().toString().length < 2 // Checking if date is < 10 and pre-prending 0 if not to adjust for date input.
+      ? `0${dateNow.getUTCDate()}`
+      : dateNow.getUTCDate();
+  
+  const materialDateInput = `${year}-${month}-${date}`;
+
+  const [values, setValues] = useState({
+    name: "",
+    phoneNumber: "",
+    dateUpdatedFrom: materialDateInput,
+    dateUpdatedUntil: materialDateInput,
+    dateRegisteredFrom: materialDateInput,
+    dateRegisteredUntil: materialDateInput,
```

Comment:
I think the whole date filtering might need a bit of a rework for the sake of uniformity with the other tables. I'd recommend taking a look at the job leads or employer tables as a reference.